code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/DnsInfoDaoImpl.java

```diff
@@ -311,4 +313,91 @@ protected DnsEntryData map(List<Object> input) {
+                .join(dnsConsumeMap)
+                .on(dnsConsumeMap.SERVICE_ID.eq(clientServiceExposeMap.SERVICE_ID))
+                .join(dnsService)
+                .on(dnsConsumeMap.CONSUMED_SERVICE_ID.eq(dnsService.ID))
+                .join(serviceConsumeMap)
+                .on(serviceConsumeMap.SERVICE_ID.eq(dnsService.ID))
+                .join(targetServiceExposeMap)
+                .on(targetServiceExposeMap.SERVICE_ID.eq(serviceConsumeMap.CONSUMED_SERVICE_ID))
+                .join(targetNic)
+                .on(targetNic.INSTANCE_ID.eq(targetServiceExposeMap.INSTANCE_ID))
+                .join(targetInstance)
+                .on(targetNic.INSTANCE_ID.eq(targetInstance.ID))
+                .join(targetNicIpTable)
+                .on(targetNicIpTable.NIC_ID.eq(targetNic.ID))
+                .join(targetIpAddress)
+                .on(targetNicIpTable.IP_ADDRESS_ID.eq(targetIpAddress.ID))
+                .join(clientNicIpTable)
+                .on(clientNicIpTable.NIC_ID.eq(clientNic.ID))
+                .join(clientIpAddress)
+                .on(clientNicIpTable.IP_ADDRESS_ID.eq(clientIpAddress.ID))
+                .join(clientInstance)
+                .on(clientNic.INSTANCE_ID.eq(clientInstance.ID))
+                .where(NIC.INSTANCE_ID.eq(instance.getId())
+                        .and(NIC.VNET_ID.isNotNull())
+                        .and(NIC.REMOVED.isNull())
+                        .and(targetIpAddress.ROLE.eq(IpAddressConstants.ROLE_PRIMARY))
+                        .and(targetIpAddress.REMOVED.isNull())
+                        .and(clientIpAddress.ROLE.eq(IpAddressConstants.ROLE_PRIMARY))
+                        .and(clientIpAddress.REMOVED.isNull())
+                        .and(targetNicIpTable.REMOVED.isNull())
+                        .and(clientNic.REMOVED.isNull())
+                        .and(targetNic.REMOVED.isNull())
+                        .and(serviceConsumeMap.REMOVED.isNull())
+                        .and(serviceConsumeMap.STATE.in(CommonStatesConstants.ACTIVATING,
+                                CommonStatesConstants.ACTIVE))
+                        .and(dnsConsumeMap.REMOVED.isNull())
+                        .and(dnsConsumeMap.STATE.in(CommonStatesConstants.ACTIVATING,
+                                CommonStatesConstants.ACTIVE))
+                        .and(clientServiceExposeMap.REMOVED.isNull())
+                        .and(targetServiceExposeMap.REMOVED.isNull())
+                        .and(targetInstance.STATE.in(InstanceConstants.STATE_RUNNING,
+                                InstanceConstants.STATE_STARTING))
+                        .and(dnsService.KIND.eq(ServiceDiscoveryConstants.KIND.DNSSERVICE.name()))
+                        .and(dnsService.STATE.in(CommonStatesConstants.ACTIVATING,
+                                CommonStatesConstants.ACTIVE)))
+                .fetch().map(mapper);
```

Comment:
@ibuildthecloud building this condition in java is even worth than typing mysql query by hand